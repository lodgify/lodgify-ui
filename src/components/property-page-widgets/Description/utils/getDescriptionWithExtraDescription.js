import React from 'react';

import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { getParagraphsFromStrings } from 'utils/get-paragraphs-from-strings';
import { Paragraph } from 'typography/Paragraph';

import { formatParagraphWithModal } from './formatParagraphWithModal';
import { isDescriptionDisplayingWithEllipsis } from './isDescriptionDisplayingWithEllipsis';

/**
 * @param  {string} descriptionText
 * @param  {string} extraDescriptionText
 * @param  {string} extraDescriptionButtonText
 * @return {Object}
 */
export const getDescriptionWithExtraDescription = (
  descriptionText,
  extraDescriptionText,
  extraDescriptionButtonText
) =>
  getParagraphsFromStrings(descriptionText).map(
    (paragraphText, index, descriptionTextArray) => (
      <Paragraph key={buildKeyFromStrings(paragraphText, index)}>
        {isDescriptionDisplayingWithEllipsis(
          index,
          descriptionTextArray,
          extraDescriptionText
        )
          ? formatParagraphWithModal(
              paragraphText,
              descriptionText,
              extraDescriptionText,
              extraDescriptionButtonText
            )
          : paragraphText}
      </Paragraph>
    )
  );
