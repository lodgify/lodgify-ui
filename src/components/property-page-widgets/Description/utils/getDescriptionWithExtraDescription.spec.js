import { getDescriptionWithExtraDescription } from './getDescriptionWithExtraDescription';

describe('getDescriptionWithExtraDescription', () => {
  it('should return the right structure', () => {
    const actual = getDescriptionWithExtraDescription('yo', 'bro', 'go');

    expect(actual).toMatchSnapshot();
  });
});
