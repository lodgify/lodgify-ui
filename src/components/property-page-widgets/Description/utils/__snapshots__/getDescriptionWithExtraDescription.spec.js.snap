// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDescriptionWithExtraDescription should return the right structure 1`] = `
Array [
  <Paragraph
    size="medium"
    weight={null}
  >
    <React.Fragment>
      yo...
      <Modal
        hasCloseIcon={true}
        hasPadding={true}
        hasRoundedCorners={false}
        header={null}
        isFullscreen={false}
        onClose={[Function]}
        size="tiny"
        trigger={
          <Button
            as="button"
            basic={true}
          >
            go
          </Button>
        }
      >
        <Paragraph
          size="medium"
          weight={null}
        >
          yo
        </Paragraph>
        <Paragraph
          size="medium"
          weight={null}
        >
          bro
        </Paragraph>
      </Modal>
    </React.Fragment>
  </Paragraph>,
]
`;
